app/javascript/controllers/partner_filter_component_controller.js

```diff
@@ -0,0 +1,71 @@
+import { Controller } from "@hotwired/stimulus";
+
+// Connects to data-controller="partner-filter-component"
+export default class extends Controller {
+	static targets = [
+		"form",
+		"category",
+		"categoryText",
+		"categoryDropdown",
+		"neighbourhood",
+		"neighbourhoodText",
+		"neighbourhoodDropdown",
+	];
+
+	connect() {
+		this.updateLabels();
+	}
+
+	submitCategory() {
+		this.submitForm();
+	}
+
+	submitNeighbourhood() {
+		this.submitForm();
+	}
+
+	resetCategory() {
+		this.selectedCategory.checked = false;
+		this.submitForm();
+	}
+
+	resetNeighbourhood() {
+		this.selectedNeighbourhood.checked = false;
+		this.submitForm();
+	}
+
+	toggleCategory() {
+		this.categoryDropdownTarget.classList.toggle("filters__dropdown--hidden");
+	}
+
+	toggleNeighbourhood() {
+		this.neighbourhoodDropdownTarget.classList.toggle(
+			"filters__dropdown--hidden"
+		);
+	}
+
+	updateLabels() {
+		// Find the associated label for each selected param and get the text contents
+		// If params are selected, they show up instead of "Category" and "Neighbourhood" text
```

Comment:
I found this confusing in the UI. but I guess it's a way to show something is selected and can only select one thing at a time from each list now, right?

Reply:
yes - i think we need to replace this whole thing with a `<select>` box - its in marks' new design but obvi getting v out of scope

Reply:
Agree. Do we bother making sure there's an issue for future us? Or will it naturally arise?

Reply:
i think its covered under #1812